api/PayrollProcessor.Core.Domain/Features/Users/UserCreateCommand.cs

```diff
@@ -0,0 +1,21 @@
+using System;
+using Ardalis.GuardClauses;
+using PayrollProcessor.Core.Domain.Intrastructure.Operations.Commands;
+
+namespace PayrollProcessor.Core.Domain.Features.Users
+{
+    public class UserCreateCommand : ICommand<User>
+    {
+        public Guid NewId { get; set; }
+        public UserNew User { get; }
+
+        public UserCreateCommand(Guid newId, UserNew user)
+        {
+            Guard.Against.Default(newId, nameof(newId));
```

Comment:
I was just doing some research into updates to FluentValidation

I think that creating validators for Queries/Commands and running the Validation as the outer most decorator for the Handlers would be a better approach to Guarding so that we could return failed Results instead of throwing exceptions.